sdk/resourcemanager/Azure.ResourceManager.Core/src/ArmException.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Runtime.Serialization;
+using Azure.Core;
+
+#nullable enable
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing exceptions thrown by Azure ARM services.
+    /// </summary>
+    [Serializable]
+    public class ArmException : RequestFailedException
+    {
+        /// <summary>
+        /// Gets and set the Code of the ARM exception.
+        /// </summary>
+        public string Code { get; internal set; }
+
+        /// <summary>
+        /// Gets and set the Target of the ARM exception.
+        /// </summary>
+        public string Target { get; internal set; }
+
+        /// <summary>
+        /// Gets and set the Details of the ARM exception.
+        /// </summary>
+        public ArmException[]? Details { get; internal set; }
+
+        /// <summary>
+        /// Gets and set the AdditionalInfo of the ARM exception.
+        /// </summary>
+        public IDictionary AdditionalInfo => Data;
```

Comment:
I don't think we generally add aliases for existing properties especially if they don't do any transformations.

Reply:
I was debating that one. Was hoping to match exactly on the ErrorResponse so user can have clear linkage between the C# exception and Rest contract. Same for the Details. Should discuss on this.